src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -245,16 +268,7 @@ bool KeyboardManagerState::DetectShortcutUIBackend(LowlevelKeyboardEvent* data)
         // Add the key if it is pressed down
         if (data->wParam == WM_KEYDOWN || data->wParam == WM_SYSKEYDOWN)
         {
-            // Set the new key and store if a change occured
-            std::unique_lock<std::mutex> lock(detectedShortcut_mutex);
-            bool updateUI = detectedShortcut.SetKey(data->lParam->vkCode);
-            lock.unlock();
-
-            if (updateUI)
-            {
-                // Update the UI. This function is called here because it should store the set of keys pressed till the last key which was pressed down.
-                UpdateDetectShortcutUI();
-            }
+            SelectDetectedShortcut(data->lParam->vkCode);

```

Comment:
The else if part also needs to be moved to that function by making some adjustments. The existing logic was that if I press Shift (down) X (down) | X (up) Shift (up), the displayed UI is only updated for the last key down, and the shortcut variable resets each key when it is released. So in this case even though at the end Shift X is displayed, the `detectedShortcut` variable will actually be empty.
With the new changes, in case of Esc and Enter we should call this second Else if part as well in the short press case, because otherwise if you press Enter (down) Enter (up) [i.e. a short press] and then you press Shift, it will show as Shift Enter instead of just Shift (coz Enter is actually released the shortcut should do ResetKey on it)

Reply:
This should be fixed in the last commit